tests/ui/did_you_mean/compatible-variants.stderr

```diff
@@ -19,9 +20,14 @@ LL +     None
 LL ~     }
 LL +     Some(())
    |
+help: consider returning a value here
+   |
+LL ~     }
+LL +     /* `Option<()>` value */
+   |
```

Comment:
Here we are providing effectively the same suggestion twice. We need to find a way to avoid one or the other (or ideally, unify the logic to a single place, particularly because the pre-existing suggestion gives you the actual value to use in this case).

Reply:
Done.

The suggestion is now invoked from `emit_type_mismatch_suggestions()` in `demand.rs` which ensures it will not be called if the other suggestion has already been emitted.

Reply:
@estebank please review